src/pe.cc

```diff
@@ -109,23 +110,37 @@ bool PeFile::Initialize() {
     return false;
   }
 
-  string_view exe_header =
-      GetRegion(dos_header_.AddressOfNewExeHeader, sizeof(pe_header));
-  memcpy(&pe_header_, exe_header.data(), exe_header.size());
+  PE_TYPE Magic;
+  auto pe_end =
+      CheckedAdd(dos_header_.AddressOfNewExeHeader, sizeof(pe_header_));
+  if (CheckedAdd(pe_end, sizeof(Magic)) > data_.size()) {
```

Comment:
@haberman I changed this code, because I also wanted to check the magic field of the optional header following the PE header.
If I understood the problem from fuzzing correctly this should still work fine?

Would it be possible for you to add a yaml file for the 'bad' PE file that the fuzzer produced?

Reply:
Every time we fix a fuzzing bug, the payload that crashed the program is added to the [fuzz corpus](https://github.com/google/bloaty/tree/master/tests/testdata/fuzz_corpus). The [`fuzz_test` test](https://github.com/google/bloaty/blob/master/CMakeLists.txt#L382-L384) will always test against every payload in the corpus.

If the code is still passing `fuzz_test`, we should be good!

We don't specify these payloads with yaml, because often they aren't valid input files at all, so `yaml2obj` wouldn't be capable of producing them.

Reply:
Great, thanks.